pkg/controller/scheduling/scheduling_utils.go

```diff
@@ -46,7 +46,8 @@ func BuildSchedulingPolicy4Component(clusterName, compName string, affinity *app
 }
 
 func buildSchedulingPolicy(cluster *appsv1alpha1.Cluster, compSpec *appsv1alpha1.ClusterComponentSpec) (*appsv1alpha1.SchedulingPolicy, error) {
-	schedulingPolicy := cluster.Spec.SchedulingPolicy
+	// make a copy, or it will affect cluster's spec
+	schedulingPolicy := cluster.Spec.SchedulingPolicy.DeepCopy()
 	if compSpec != nil && compSpec.SchedulingPolicy != nil {
 		schedulingPolicy = compSpec.SchedulingPolicy
```

Comment:
Does this line need a deep copy as well?

Reply:
looks like compSpec is either deep copied or created from scratch when created:
https://github.com/apecloud/kubeblocks/blob/5d60cde75a033b0e42c45c4b51a8f8fa4e63ec1c/controllers/apps/transformer_cluster_api_normalization.go#L147-L152

https://github.com/apecloud/kubeblocks/blob/5d60cde75a033b0e42c45c4b51a8f8fa4e63ec1c/controllers/apps/transformer_cluster_api_normalization.go#L109-L116

Reply:
So what @zjx20 means is that though compSpec is deepcopyed, it should remain unchanged in this function.